src/screens/Events/Events.tsx

```diff
@@ -83,69 +106,39 @@ const Events: React.FC<EventsProps> = ({ currentUser, setCurrentEventId }) => {
         />
       </View>
       <View style={styles.eventList}>
-        <Text style={styles.text}>Upcoming</Text>
-        {!loading &&
-          events &&
-          events.accepted.map((foodDrive: any) => (
-            <ListItem
-              id={`${foodDrive.id}-list-item`}
-              key={`${foodDrive.id}-list-item`}
-              title={foodDrive.event.name}
-              otherInformation={`${dayjs(foodDrive.event.startDate).format(
-                "DD MMM YY h:mm A",
-              )} - ${dayjs(foodDrive.event.endDate).format(
-                "DD MMM YY h:mm A",
-              )}`}
-              onPress={() => {
-                setCurrentEventId(foodDrive.event.id);
-                navigation.navigate(SCREEN_NAMES.common.events.eventDetails);
-              }}
-              iconColor={colors.success}
-            />
-          ))}
-      </View>
-      <View style={styles.eventList}>
-        <Text style={styles.text}>Pending confirmation</Text>
-        {!loading &&
-          events &&
-          events.pending.map((foodDrive: any) => (
-            <ListItem
-              id={`${foodDrive.id}-list-item`}
-              key={`${foodDrive.id}-list-item`}
-              title={foodDrive.event.name}
-              otherInformation={`${dayjs(foodDrive.event.startDate).format(
-                "DD MMM YY h:mm A",
-              )} - ${dayjs(foodDrive.event.endDate).format(
-                "DD MMM YY h:mm A",
-              )}`}
-              onPress={() => {
-                setCurrentEventId(foodDrive.event.id);
-                navigation.navigate(SCREEN_NAMES.common.events.eventDetails);
-              }}
-              iconColor={colors.orange}
-            />
-          ))}
-      </View>
-      <View style={styles.eventList}>
-        <Text style={styles.text}>Rejected events</Text>
         {!loading &&
-          events &&
-          events.rejected.map((foodDrive: any) => (
-            <ListItem
-              id={`${foodDrive.id}-list-item`}
-              key={`${foodDrive.id}-list-item`}
-              title={foodDrive.event.name}
-              otherInformation={`${dayjs(foodDrive.event.startDate).format(
-                "DD MMM YY h:mm A",
-              )} - ${dayjs(foodDrive.event.endDate).format(
-                "DD MMM YY h:mm A",
-              )}`}
-              onPress={() => {
-                setCurrentEventId(foodDrive.event.id);
-                navigation.navigate(SCREEN_NAMES.common.events.eventDetails);
-              }}
-            />
-          ))}
+          invitations &&
+          Object.entries(InvitationStatusDescriptions).map(
```

Comment:
React Native has a component that's used for displaying sectioned lists. Made this more generic in our component library.Have a look at the SectionedList component. Much preferred to use that one instead of building it from scratch

Reply:
For the sectioned list component, you have to create data that's an array of title -> data fields